fdbclient/NativeAPI.actor.cpp

```diff
@@ -1977,6 +1979,36 @@ Future<Optional<Value>> Transaction::get( const Key& key, bool snapshot ) {
 	if( !snapshot )
 		tr.transaction.read_conflict_ranges.push_back(tr.arena, singleKeyRange(key, tr.arena));
 
+	if(key == metadataVersionKey) {
+		if(!ver.isReady() || metadataVersion.isSet()) {
```

Comment:
A knowledge question (Not related to code review): 
Under which situation, the getVersion() will return a ver that is not ready?
If metadataVersion (a promise) is not set, what metadataVersion.getFuture() will return, a Void or a brokenPromise?

Reply:
If you call get() in a transaction before getting a read version, the get() will wait for a read version before doing the get().
metadataVersion.getFuture() will return a future that will be ready when metadataVersion.set() is called. If metadataVersion is destroyed while you are still waiting on metadataVersion.getFuture(), you will get a broken_promise error.